libs/semanticsearch/include/semanticsearch/location.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "semanticsearch/index/base_types.hpp"
+
+#include <memory>
+#include <string>
+
+namespace fetch {
+namespace semanticsearch {
+
+struct Location
+{
+  std::string      model;
+  SemanticPosition position;
+  bool             operator<(Location const &other) const
```

Comment:
If you add a newline above and then run `clang-format` it won't insist on this crazy formatting 😁